library/src/main/java/com/nextcloud/android/lib/resources/files/ToggleFileLockRemoteOperation.kt

```diff
@@ -0,0 +1,91 @@
+ *
+ * @author Álvaro Brey Vilas
+ * Copyright (C) 2022 Álvaro Brey Vilas
+ * Copyright (C) 2022 Nextcloud GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.nextcloud.android.lib.resources.files
+
+import com.nextcloud.common.NextcloudClient
+import com.nextcloud.common.OkHttpMethodBase
+import com.nextcloud.operations.LockMethod
+import com.nextcloud.operations.UnlockMethod
+import com.owncloud.android.lib.common.accounts.AccountUtils
+import com.owncloud.android.lib.common.network.WebdavUtils
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.utils.Log_OC
+import org.apache.commons.httpclient.HttpStatus
+
+/**
+ * Favorite or unfavorite a file.
+ *
+ * @param toLock `true` if file is to be locked, `false` if it is to be unlocked
+ */
+class ToggleFileLockRemoteOperation(private val toLock: Boolean, private val filePath: String) :
+    RemoteOperation<Void>() {
+    @Suppress("TooGenericExceptionCaught")
+    override fun run(client: NextcloudClient): RemoteOperationResult<Void> {
+
+        var method: OkHttpMethodBase? = null
+        var result: RemoteOperationResult<Void>?
+
+        try {
+            // remote request
+            val uri = getUri(client)
+            method = when (toLock) {
+                true -> LockMethod(uri, false)
+                false -> UnlockMethod(uri, false)
+            }
+            method.addRequestHeader(HEADER_USER_LOCK, HEADER_USER_LOCK_VALUE)
+
+            val status: Int = client.execute(method)
+            if (isSuccess(status)) {
+                result = RemoteOperationResult(true, method)
+            } else {
+                result = RemoteOperationResult(false, method)
+                method.releaseConnection()
+            }
+        } catch (e: Exception) {
+            result = RemoteOperationResult(e)
+            Log_OC.e(TAG, "Lock toggle failed: " + result.logMessage, result.exception)
+        } finally {
+            method?.releaseConnection()
+        }
+
+        return result!!
+    }
+
+    private fun getUri(client: NextcloudClient): String {
+        return client.baseUri.toString() + AccountUtils.WEBDAV_PATH_9_0 +
+            "/files/" + client.getUserIdEncoded() + "/" + WebdavUtils.encodePath(filePath)
```

Comment:
This is in OwnCloudClient right now.
To not duplicate this, we can either 
- extract it to a helper class
- put this one in NextcloudClient

Having it only here feels not good :)

Reply:
Done https://github.com/nextcloud/android-library/pull/851/commits/80acd8b5bc1a363f5444f338a32fd133dbc7c966 and https://github.com/nextcloud/android-library/pull/851/commits/890762cf2767e695ab8a50604c3426fdd92be6a4

I used a delegate, since it also holds state (baseUri, userId) and I couldn't use an abstract class because OwncloudClient already has a parent